### PR TITLE
Update ci to newer python versions

### PR DESCRIPTION
**Pull request by albi3ro**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,8 +4,12 @@
 
 ### Breaking changes
 
+* Removes testing for Python 3.7.
```
Comment (albi3ro) [inline CHANGELOG.md:7]:
```suggestion
* Removes testing for Python 3.7.
  [(#237)](https://github.com/PennyLaneAI/pennylane-qiskit/pull/237)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,8 +4,12 @@
...
+
 ### Improvements
 
+* Adds testing for Python 3.11.
```
Comment (albi3ro) [inline CHANGELOG.md:11]:
```suggestion
* Adds testing for Python 3.11.
  [(#237)](https://github.com/PennyLaneAI/pennylane-qiskit/pull/237)
```

**Review by timmysilv — DISMISSED** [review]

**Comment by AlbertMitjans** [comment]
I encountered this sphinx error before! It seems like our sphinx version doesn't recognise walrus operators.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9, '3.10']
+        python-version: [3.8, 3.9, '3.10', '3.11']
```
Comment (AlbertMitjans) [inline .github/workflows/tests.yml:14]: The python version of the `integration tests` also needs to be changed.

**Review by AlbertMitjans — APPROVED** [review]
Approved!

**Pull request merged**